src/messages/messages.ts

```diff
@@ -99,3 +102,48 @@ export const createReactionMessage = (fromId: string, emoji: string, inReplyTo:
   fromId,
   inReplyTo,
 });
+
+/**
+ * hash() takes a DSNPMessage and returns a standard hash for use in signatures
+ * as defined in the [Message Signatures](https://github.com/LibertyDSNP/spec/blob/main/pages/Messages/Signatures.md)
+ * specification.
+ *
+ * @param   message The message to hash
+ * @returns         A Uint8Array of the resulting hash
+ */
+export const hash = (message: DSNPMessage): string => {
+  const sortedMessage = sortObject((message as unknown) as Record<string, unknown>);
+  let serialization = "";
+
+  for (const key in sortedMessage) {
+    serialization = `${serialization}${key}${sortedMessage[key]}`;
+  }
+
+  return keccak256(serialization);
+};
+
+/**
+ * sign() takes a private key and DSNP message and returns a Uint8Array
+ * signature for the message.
+ *
+ * @param message The DSNP message to sign
+ * @param opts    Optional. Configuration overrides, such as from address, if any
+ * @returns       The message signature as a Uint8Array
+ */
+export const sign = async (message: DSNPMessage, opts?: Config): Promise<string> => {
+  const { signer } = await getConfig(opts);
+  if (!signer) throw MissingSigner;
+  return signer.signMessage(hash(message));
+};
+
+/**
+ * verify() takes a public key, DSNP message and a message signature and returns
+ * whether the signature is valid for the given key and message.
+ *
+ * @param message   The DSNP message to sign
+ * @param signature The message signature to validate
+ * @returns         The address of the signer
+ */
+export const verify = (message: DSNPMessage, signature: string): string => {
+  return ethers.utils.verifyMessage(hash(message), signature);
```

Comment:
I think we want to use `ethers.utils.recoverPublicKey` here to prevent using all the special Ethereum stuff we don't need.

Reply:
**Notes following out-of-band conversation:**

- There's no way around the Ethereum's RPC prefix
- We have to update the spec to use the prefix instead
- PR to do so here: https://github.com/LibertyDSNP/spec/pull/68